# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/xul/XULPopupElement.cpp
# Commit: 3a8608560898
# Full Hash: 3a86085608988058ad0774d0878e5aedfea61f49
# Author: stransky <stransky@redhat.com>
# Date: 2022-11-09 09:58:36
# Regressor Bug: 1761877
# File Overlap Count: 1
# Description:
#   Bug 1761877 [Linux] Add isWaylandDragSource and isWaylandPopup attributes to XULPopupElement r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161127
# ==============================================================================

diff -r 02c859d135b7 -r 3a8608560898 dom/xul/XULPopupElement.cpp
--- a/dom/xul/XULPopupElement.cpp	Tue Nov 08 20:27:50 2022 +0000
+++ b/dom/xul/XULPopupElement.cpp	Tue Nov 08 20:43:17 2022 +0000
@@ -17,6 +17,9 @@
 #include "mozilla/dom/Event.h"
 #include "mozilla/dom/XULPopupElement.h"
 #include "mozilla/dom/XULPopupElementBinding.h"
+#ifdef MOZ_WAYLAND
+#  include "mozilla/WidgetUtilsGtk.h"
+#endif
 
 namespace mozilla::dom {
 
@@ -273,4 +276,21 @@
   }
 }
 
+bool XULPopupElement::IsWaylandDragSource() const {
+#ifdef MOZ_WAYLAND
+  nsMenuPopupFrame* menuPopupFrame = do_QueryFrame(GetPrimaryFrame());
+  return menuPopupFrame->IsDragSource();
+#else
+  return false;
+#endif
+}
+
+bool XULPopupElement::IsWaylandPopup() const {
+#ifdef MOZ_WAYLAND
+  return widget::GdkIsWaylandDisplay();
+#else
+  return false;
+#endif
+}
+
 }  // namespace mozilla::dom